File: tv_android/app/src/main/java/com/nascabos/tv/core/util/ServerVersionUtil.kt
package com.nascabos.tv.core.util

object ServerVersionUtil {
    fun parseMajorVersion(version: String?): Int? {
        val trimmed = version?.trim().orEmpty()
        if (trimmed.isEmpty()) return null
        val normalized =
            if (trimmed.startsWith("v", ignoreCase = true)) {
                trimmed.substring(1)
            } else {
                trimmed
            }
        val parts = normalized.split('.')
        if (parts.isEmpty()) return null
        val first = parts.first()
        val match = Regex("(\\d+)").find(first) ?: return null
        return match.groupValues.getOrNull(1)?.toIntOrNull()
    }

    fun isAtLeast(
        version: String?,
        majorVersion: Int,
        unknownAsSupported: Boolean = true,
    ): Boolean {
        val major = parseMajorVersion(version)
        if (major == null) return unknownAsSupported
        return major >= majorVersion
    }
}

File: flutter_client/lib/utils/server_version_util.dart
class ServerVersionUtil {
  const ServerVersionUtil._();

  static int? parseMajorVersion(String? version) {
    if (version == null) return null;
    final trimmed = version.trim();
    if (trimmed.isEmpty) return null;
    final normalized = (trimmed.startsWith('v') || trimmed.startsWith('V'))
        ? trimmed.substring(1)
        : trimmed;
    final parts = normalized.split('.');
    if (parts.isEmpty) return null;
    final match = RegExp(r'(\d+)').firstMatch(parts.first);
    if (match == null) return null;
    return int.tryParse(match.group(1)!);
  }

  static bool isAtLeast(
    String? version,
    int majorVersion, {
    bool unknownAsSupported = true,
  }) {
    final major = parseMajorVersion(version);
    if (major == null) return unknownAsSupported;
    return major >= majorVersion;
  }
}
